### PR TITLE
Add support in Commons CSV for tracking byte positions during parsing

### PR DESCRIPTION
**Pull request by DarrenJAN**
Please note that it is required to use Java 11 to build and run tests.

Here are summaries of the modifications:
1. Add our test data files to rat exclusions to avoid unapproved license checks in pom.xml
2. Add CSVParser constructor
3. Add CSVRecord constructor
    1. Remove mapping Since version 1.8, the mapping between the column header and the column index was removed from the serialized state (Note: Commons CSV: Latest version is 1.12.0)
4. Add other Parser helper functions
    1. ExtendedBufferedReader()
    2. getBytesRead()
    3. getCharacterByte() in CSVRecord class
5. Add new Junit test — src/test/java/org/apache/commons/csv/issues/JiraCsv196Test.java
   1. Update deprecated import package name (ie: import org.junit.Test; - > import org.junit.jupiter.api.Test; ; import org.junit.Assert →  import static org.junit.jupiter.api.Assertions.assertEquals; )
   2. new InputStreamReader object to get test input
6. Override the peek() functions in ExtendedBufferedReader class to avoid consuming the byte 

Test result:
mvn 
image
image
Pass unit tests and other restrictions

### CODE REVIEW

File: src/main/java/org/apache/commons/csv/CSVFormat.java [hunk 1]
```diff
@@ -2074,6 +2074,25 @@ public CSVParser parse(final Reader reader) throws IOException {
...
     }
 
+    /**
+     * Parses the specified content.
+     *
+     * <p>
+     * See also the various static parse methods on {@link CSVParser}.
+     * </p>
+     *
+     * @param reader the input stream
```
Comment (vzhang-progress) [inline src/main/java/org/apache/commons/csv/CSVFormat.java:2089]: Please improve the comments to something that makes more sense.

File: src/main/java/org/apache/commons/csv/CSVParser.java [hunk 2]
```diff
@@ -438,10 +438,15 @@ public CSVParser(final Reader reader, final CSVFormat format) throws IOException
     @SuppressWarnings("resource")
     public CSVParser(final Reader reader, final CSVFormat format, final long characterOffset, final long recordNumber)
         throws IOException {
+            this(reader, format, characterOffset, recordNumber, null);
+        }
+
+    public CSVParser(final Reader reader, final CSVFormat format, final long characterOffset, final long recordNumber,
```
Comment (vzhang-progress) [inline src/main/java/org/apache/commons/csv/CSVParser.java:468]: Please add comments here, you can follow the comments for other parsers.

**Review by yunzvanessa — CHANGES_REQUESTED** [review]
Hi Yuzhan,

The functionality looks good. Two things to focus on:
1. Please go over all the comments you added since we would like to formalize it and attempt for a PR against the central repo.
2. Please confirm the copyright info with Shawna and update them in the source files you are changing accordingly.

**Comment by vzhang-progress** [comment]
Can you also investigate the failing checks below and make sure they are okay?

**Comment by DarrenJAN** [comment]
> Hi Yuzhan,
> 
> The functionality looks good. Two things to focus on:
> 
> 1. Please go over all the comments you added since we would like to formalize it and attempt for a PR against the central repo.
> 2. Please confirm the copyright info with Shawna and update them in the source files you are changing accordingly.

1. I updated some comments
2. Should I confirm the copyright right now or wait until creating PR for the upstream project?

File: src/main/java/org/apache/commons/csv/CSVParser.java [hunk 3]
```diff
@@ -441,6 +441,31 @@ public CSVParser(final Reader reader, final CSVFormat format, final long charact
...
+     * @param recordNumber
+     *            The next record number to assign
+     * @param encoding
+     *            The encoding to use for the reader
+     * @throws IllegalArgumentException
+     *             If the parameters of the format are inconsistent or if either the reader or format is null.
+     * @throws IOException
+     *             If there is a problem reading the header or skipping the first record
+     * @throws CSVException Thrown on invalid input.
+     * @since 1.12
```
Comment (vzhang-progress) [inline src/main/java/org/apache/commons/csv/CSVParser.java]: Please remove "since 1.12", since commons-csv will decide in which version it will be in

**Review by yunzvanessa — CHANGES_REQUESTED** [review]
Hi Yuzhan,

Just a minor detail for you to change. Thanks

**Pull request merged**